_source/_posts/2022-04-21-jhipster-k8s-azure.md

```diff
@@ -0,0 +1,667 @@
+- [Docker Hub](https://hub.docker.com/): you'll need this to host the docker images so that Azure can pull them
+- [Java 11](https://adoptopenjdk.net/)+
+- [Okta CLI](https://cli.okta.com/manual/#installation): you'll use Okta to add security to the microservice network. You can register for a free account from the CLI.
+- [Azure Cloud account](https://azure.microsoft.com/en-us/free/): they offer an account with a $200 credit to start out. 
+- [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli): you'll use the Azure CLI to manage the Kubernetes cluster
+- [kubectl](https://kubernetes.io/docs/tasks/tools/): CLI to manage Kubernetes clusters
+
+#### Modifying the Generated JHipster Microservice Project for Azure and Cosmos DB
+
+This project is based on two of Matt Raible's tutorials: *[Reactive Java Microservices with Spring Boot and JHipster](https://developer.okta.com/blog/2021/01/20/reactive-java-microservices)* and *[Kubernetes to the Cloud with Spring Boot and JHipster](https://developer.okta.com/blog/2021/06/01/kubernetes-spring-boot-jhipster)*. In these tutorials, he builds a reactive Java microservice and shows how to deploy it to Google Cloud (GCP). I have modified the project to work with Azure and Cosmos DB.
+
+You will first run the project using Docker Compose. Once you have this working, you will run the project as a Kubernetes cluster on Azure. The modifications were relatively minor and involved removing the unnecessary MongoDB instances (from both the `docker-compose.yml` file and from the Kubernetes descriptors) as well as updating environment values to point the `store` service to the Cosmos DB instance instead of a MongoDB instance.
+
+Listed briefly, the changes I made from Matt Raible's posts to make this work with Azure and Cosmos DB are as follows. You can skip this list and go right to cloning the Git repository if you want, but since this documents how to update a JHipster-generated project to use Cosmos DB, I thought it was worth putting in here.
+
+In the `docker-compose/docker-compose.yml`  file:
+
+- removed the MongoDB service
+- updating the `store` service property `SPRING_DATA_MONGODB_URI` to point to the Cosmos DB instance via a `.env` file
+- removed the Keycloak service and the environment variables that configured auth to use keycloak from the remaining services (not strictly necessary but cleaned things up)
+
+In the `k8s/store-k8s` directory:
+
+- removed the `store-mongodb.yml` file (this creates the MongoDB Kubernetes service that this project does not need)
+- in `store-deployment.yml`
+  - removed the `initContainers`  (the init container waits for the MongoDB instance, which is removed)
+  - updated `SPRING_DATA_MONGODB_URI` env value of the `store-app` container to the Cosmos DB URI (points the store to the Cosmos DB instance)
+  - properly secure the Cosmos DB connection string using Kubernetes secrets and `kubeseal`
```

Comment:
```suggestion
  - properly secured the Cosmos DB connection string using Kubernetes secrets and `kubeseal`
```
Because the rest of the items use past tense.